components/timezone/src/time_zone.rs

```diff
@@ -40,8 +96,67 @@ impl TimeZoneInfo {
         Self {
             offset: Some(UtcOffset::zero()),
             time_zone_id: TimeZoneBcp47Id(tinystr::tinystr!(8, "utc")),
-            zone_variant: Some(ZoneVariant::standard()),
-            local_time: None,
+            zone_variant: (),
+            local_time: (),
+        }
+    }
+
+    /// Creates a new [`TimeZoneInfo`] with the given offset and time zone ID.
+    pub const fn from_id_and_offset(time_zone_id: TimeZoneBcp47Id, offset: UtcOffset) -> Self {
+        Self {
+            offset: Some(offset),
+            time_zone_id,
+            zone_variant: (),
+            local_time: (),
+        }
+    }
+
+    /// Sets a local time on this time zone.
+    pub const fn with_local_time(
+        self,
+        local_time: (Date<Iso>, Time),
+    ) -> TimeZoneInfo<models::AtTime> {
+        TimeZoneInfo {
+            offset: self.offset,
+            time_zone_id: self.time_zone_id,
+            zone_variant: (),
+            local_time,
+        }
+    }
+}
+
+impl TimeZoneInfo<models::AtTime> {
+    /// Sets a zone variant on this time zone.
+    pub const fn with_zone_variant(self, zone_variant: ZoneVariant) -> TimeZoneInfo<models::Full> {
+        TimeZoneInfo {
+            offset: self.offset,
+            time_zone_id: self.time_zone_id,
+            zone_variant,
+            local_time: self.local_time,
+        }
+    }
+}
+
+impl TimeZoneInfo<models::Full> {
+    /// Creates a new [`TimeZoneInfo`] for the UTC time zone
+    /// with a reference point at the UNIX Epoch.
```

Comment:
mmh kinda don't want to document this. The reference point for UTC is completely irrelevant, because its metazone doesn't change.

Reply:
I'll delete this function since it has only 1 call site.

Reply:
I think it's still useful to have this as a public constructor

Reply:
No actually I think this is a terrible function. For example, someone might make a `TimeZoneInfo::utc_full` as the only non-string constructor of `TimeZoneInfo<Full>` and then try to overwrite fields on it.

This function clearly is not that useful because there is only 1 call site and it is a weird call site in a match statement. There are no call sites in normal code.